test/e2e/page-objects/pages/homepage.ts

```diff
@@ -0,0 +1,168 @@
+import { strict as assert } from 'assert';
+import { Driver } from '../../webdriver/driver';
+import { DEFAULT_GANACHE_ETH_BALANCE_DEC } from '../../constants';
+import HeaderNavbar from './header-navbar';
+
+class HomePage {
+  private driver: Driver;
+
+  private sendButton: string;
+
+  private activityTab: string;
+
+  private tokensTab: string;
+
+  private balance: string;
+
+  private completedTransactions: string;
+
+  private confirmedTransactions: object;
+
+  private transactionAmountsInActivity: string;
+
+  public headerNavbar: HeaderNavbar;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+    this.headerNavbar = new HeaderNavbar(driver);
+    this.sendButton = '[data-testid="eth-overview-send"]';
+    this.activityTab = '[data-testid="account-overview__activity-tab"]';
+    this.tokensTab = '[data-testid="account-overview__asset-tab"]';
+    this.confirmedTransactions = {
+      text: 'Confirmed',
+      css: '.transaction-status-label--confirmed',
+    };
+    this.balance = '[data-testid="eth-overview__primary-currency"]';
+    this.completedTransactions = '[data-testid="activity-list-item"]';
+    this.transactionAmountsInActivity =
+      '[data-testid="transaction-list-item-primary-currency"]';
+  }
+
+  async check_pageIsLoaded(): Promise<void> {
+    try {
+      await this.driver.waitForSelector(this.sendButton);
+      await this.driver.waitForSelector(this.activityTab);
+      await this.driver.waitForSelector(this.tokensTab);
+    } catch (e) {
+      console.log('Timeout while waiting for home page to be loaded', e);
+      throw e;
+    }
+    console.log('Home page is loaded');
+  }
+
+  async check_expectedBalanceIsDisplayed(
+    expectedBalance: string = DEFAULT_GANACHE_ETH_BALANCE_DEC,
+  ): Promise<void> {
+    try {
+      await this.driver.waitForSelector({
+        css: this.balance,
+        text: `${expectedBalance} ETH`,
+      });
+    } catch (e) {
+      const actualBalance = await this.getBalance();
+      const errorMessage = `Expected balance ${expectedBalance} ETH, got balance ${actualBalance} ETH`;
+      console.log(errorMessage, e);
+      throw e;
+    }
+    console.log(
+      `Expected balance ${expectedBalance} ETH is displayed on homepage`,
+    );
+  }
+
+  async getBalance(): Promise<number> {
+    const element = await this.driver.findElement(this.balance);
+    await this.driver.waitForNonEmptyElement(element);
+    return parseFloat(await element.getText());
+  }
```

Comment:
This is a dangerous pattern that can cause race conditions and flakiness.  You want to search for the text you expect to find, not just get the current text and compare it.

Reply:
Nice catch! I agree it can cause flakiness. Initially, it was used in the function `check_expectedBalanceIsDisplayed()`. When the expected balance is not displayed, we call it to get the actual balance and log it in the error message. Your comment makes me realized that people can call it separately, which indeed will cause flakiness. I have removed this function, so we will force people to use `check_expectedBalanceIsDisplayed()`.